### ISSUE TITLE
Sentinel? tilt on master failure

### ISSUE THREAD

**Issue by tunurgitr** [microsoftarchive/redis#48]
My symptoms are this:

When running Sentinel, any time I shut down the master, I get "tilt" errors on each of the sentinels.  I can shut down the Slaves with no problem.

I'm running all my instances on the same PC (I've tried other configurations too, but this is the simplest one to describe).

My config files are:
### Master

```
port 6379
timeout 0
```
### Slave

```
port 6380
timeout 0
slaveof 127.0.0.1 6379
slave-serve-stale-data yes
repl-ping-slave-period 10
repl-timeout 60
```
### Sentinel01

```
port 26379
sentinel monitor mymaster 127.0.0.1 6379 2
sentinel down-after-milliseconds mymaster 5000
sentinel failover-timeout mymaster 900000
sentinel can-failover mymaster yes
sentinel parallel-syncs mymaster 1
```
### Sentinel02

```
port 26380
sentinel monitor mymaster 127.0.0.1 6379 2
sentinel down-after-milliseconds mymaster 5000
sentinel failover-timeout mymaster 900000
sentinel can-failover mymaster yes
sentinel parallel-syncs mymaster 1
```
### Sentinel03

```
port 26381
sentinel monitor mymaster 127.0.0.1 6379 2
sentinel down-after-milliseconds mymaster 5000
sentinel failover-timeout mymaster 900000
sentinel can-failover mymaster yes
sentinel parallel-syncs mymaster 1
```

**Comment by HenryRawas** [comment]
I just made a fix for issue #43 which I believe also addresses this issue. Please try it out.

**Comment by BeatriceOltean** [comment]
Investigating if this is still an active issue

**Comment by plakhlani** [comment]
I downloaded recent branch 2_6_14 and tested it with same configuration stated above with x64 windows 7 on same machine. Instead of getting tilt mode entered, now I am getting -cmd-link-reconnection-master when master gets down. Now I wanted to know:
[1] How can I configure sentinel to promote slave1/slave2 to become new master?
[2] How can I configure sentinel to tell slave 2 become the slave of slave1 or visa versa once [1] is done?

Is it possible to do above 2 things with current release?

# Updates - 10/9/2013

I know now the answer of above two. I am now using release 2.6.12 and Redis64 in bin is not working for me but Redis is working properly. I guess this version deson't reconfigure to mymaster back when its up after down but that's already resolved in upcoming releases. So all good. 

The only question is: I am using windows 7 x64, and Sentinel in Redis64 is not working but works in 32bit.

I hope this helps.

**Comment by jepickett** [comment]
2.8.4 and 2.6.14 have been released. The signed binaries are available from NuGet and Chocolatey. Please test with these versions and let us know if the issue is resolved.

**Comment by jepickett** [comment]
I have just checked in a fix into the 2.8.4_msopen branch that appears to resolve an issue with sentinel mode. Please test with the version contained bin\release\redis-2.8.4.zip. I will close this issue in a week if there is no further feedback.
